### ISSUE TITLE
Strip Lnull; from LVTs

### ISSUE THREAD

**Issue by Siphalor** [SpongePowered/Mixin#386]
Under some weird circumstances (called OptiFine) something like
```
aconst_null
astore_2
```
sneaks into the bytecode.
This causes this in the LVT:
```
/*   LOCAL                  TYPE  NAME                                                                                                                                                                                      */
/*   [  0]             class_315  this                                                                                                                                                                                      */
/* > [  1]           PrintWriter  printwriter                                        <capture>                                                                                                                              */
/*   [  2]                  null  var2                                               <capture>         
[...]                                                                                                                    
```

It would be cool if these `Lnull;` LVT entries would get stripped so injections with local captures would be possible 😉

Or at least make it possible to `@Coerce` `Lnull;` to `Ljava/lang/Object;` (weird - but ok 🤔).

**Comment by Mumfrey** [comment]
I'm not sure I follow. ACONST_NULL shouldn't result in `Lnull;` like that, I'd have to look into it because that's not intended behaviour. In general the type of reference variables is obtained from the LVT (if one is available) so the correct variable type should be detected.

Do you have a test case I can use to replicate this, ideally a simple mixin which fails when optifine is in use.

**Comment by Siphalor** [comment]
That one: https://github.com/Siphalor/amecs/blob/master/src/main/java/de/siphalor/amecs/mixin/MixinGameOptions.java#L24

**Comment by Siphalor** [comment]
And that's the dumped bytecode of the patched class: https://hatebin.com/zsfcogcssb

**Comment by Siphalor** [comment]
For reference that's the `javap` lvt:
```

      LocalVariableTable:
        Start  Length  Slot  Name   Signature
         1958      64     7 keyString   Ljava/lang/String;
         1972      50     8 keyModifier   Ljava/lang/Object;
         1980      42     9 keyModifierNone   Ljava/lang/Object;
         1909     158     6 keybinding   Lnet/minecraft/class_304;
         2097      43     6 soundcategory   Lnet/minecraft/class_3419;
         2170      48     6 playermodelpart   Lnet/minecraft/class_1664;
           52    2243     1 printwriter   Ljava/io/PrintWriter;
         2299      12     1 exception   Ljava/lang/Exception;
            0    2320     0  this   Lnet/minecraft/class_315;
```

**Comment by TelepathicGrunt** [comment]
I have found a case in 1.16.3 Fabric where doing an inject mixin into an anonymous class's method caused the Lnull; issue with local captures. Specifically, I am trying to mixin to net.minecraft.util.dynamic.RegistryOps$EntryLoader$1's load method and local capture at decoder.parse. There are two `Throwable var10 = null;` that are turned into Lnull; in the capture and prevents me from capturing JsonElement and adding more info to latest.logs for broken worldgen datapacks.

Here's what the LocalCapture.PRINT shows and where I am trying to mixin into : https://hatebin.com/udfanhndsp
I'm trying to get the 12th local (JsonElement) but locals 8 and 10 are preventing the mixin from applying with a Critical injection failure. This error shows those locals became Lnull; https://hatebin.com/finjzzqojf

Here's my mixin setup:
https://hatebin.com/lwnhqhglcd

**Comment by Mumfrey** [comment]
Thanks for the detail, I discovered the root cause of this a week ago with the help of some peeps in the forge discord. It turns out that the issue occurs when a variable is declared and initialised to `null` before it's assigned a real value as you say. I can now catch this situation and return a better result (basically `Object` because we know it'll be `null` anyway).

**Comment by Mumfrey** [comment]
Fixed in 0.8.3
